index.html

```diff
@@ -660,39 +695,71 @@ <h2>Queue a <code>PerformanceEntry</code></h2>
+        associated with the <a
+        data-cite="HTML/webappapis.html#concept-relevant-global">relevant
+        global object</a> of <a
+        data-cite="DOM#context-object">context object</a>.</li>
+        <li>Let <var>tuple list</var> be an empty <a
+        data-cite="WHATWG-INFRA#list">list</a>.</li>
+        <li>If <var>type</var> is not null, append the result of <a
+        data-cite="WHATWG-INFRA#map-get">getting the value of entry</a> on
+        <var>map</var> given <var>type</var> as <a
+        data-cite="WHATWG-INFRA#map-key">key</a> to <var>tuple list</var>.
+        Otherwise, assign the result of <a
+        data-cite="WHATWG-INFRA#map-getting-the-values">get the values</a> on <var>map</var> to
+        <var>tuple list</var></li>
+        <li>For each <var>tuple</var> in <var>tuple list</var>, run the
+        following steps:
           <ol>
-            <li>If <var>name</var> is not <code>null</code> and
-            <var>entryObject</var>'s `name` attribute does not match
-            <var>name</var> in a <a data-cite=
-            "HTML/infrastructure.html#case-sensitive">case-sensitive</a>
-            manner, go to next <var>entryObject</var>.
-            </li>
-            <li>If <var>type</var> is not <code>null</code> and
-            <var>entryObject</var>'s `type` attribute does not match
-            <var>type</var> in a <a data-cite=
-            "HTML/infrastructure.html#case-sensitive">case-sensitive</a>
-            manner, go to next <var>entryObject</var>.
-            </li>
-            <li>Add <var>entryObject</var> to the <var>list of entry objects</var>.
-            </li>
+            <li>Let <var>buffer</var> be <var>tuple</var>'s <a>performance
+            entry buffer</a>.</li>
+            <li>If <var>tuple</var>'s <a>availableFromTimeline</a> is false,
+            continue to the next <var>tuple</var>.</li>
+            <li>Let <var>entries</var> be the result of running <a>filter
+            buffer by name and type</a> with <var>buffer</var>, <var>name</var>
+            and <var>type</var> as inputs.</li>
+            <li>For each <a>PerformanceEntry</a> <var>entry</var> in
+            <var>entries</var>, <a
+            data-cite="WHATWG-INFRA#list-append">append</a> it to
+            <var>result</var>.</li>
           </ol>
         </li>
-        <li>Return the <var>list of entry objects</var>.
+        <li>Sort <var>results</var>'s entries in chronological order with
+        respect to <a>startTime</a></li>
+        <li>Return <var>result</var>.</li>
+      </ol>
+      <p>When asked to run the <dfn>filter buffer by name and type</dfn>
+      algorithm, with <var>buffer</var>, <var>name</var>, and <var>type</var>
+      as inputs, run the following steps:</p>
+      <ol>
+        <li>Let <var>result</var> be an initially empty <a
+        data-cite="WHATWG-INFRA#list">list</a>.</li>
+        <li>Sort <var>buffer</var>'s entries in chronological order with
```

Comment:
I think it's better to sort the results list here, rather than the underlying buffer.

Reply:
To better explain my comment, which I now see is not very clear. It would be better to move the sorting step until after the loop is run on each entry in the buffer, and then sort `results`

Reply:
Done.